### ISSUE TITLE
connectionLost in examples is omitting reason arg

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1818]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)| @jerub reported|
|-|-|
|Trac ID|trac#1818|
|Type|defect|
|Created|2006-06-14 05:25:18Z|

r17198 should fix this, and I reviewed that code and it was okay, but should point out:

```
$  grep -r "connectionLost(self)" . | grep -v .svn
./twisted/words/im/proxyui.py:    def connectionLost(self):
./twisted/mail/test/test_mail.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/mail.py:    def connectionLost(self):
./twisted/mail/smtp.py:    def connectionLost(self):
./twisted/spread/pb.py:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-5.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-6.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-7.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-8.tac:    def connectionLost(self):
./doc/mail/examples/emailserver.tac:    def connectionLost(self):
./doc/historic/2002/ipc10/twisted-network-framework/index.html:    def connectionLost(self): sel                    f.peer.loseConnection();del self.peer
./doc/mail/tutorial/smtpserver/smtpserver-5.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-6.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-7.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-8.tac:    def connectionLost(self):
./doc/mail/examples/emailserver.tac:    def connectionLost(self):
./doc/historic/2002/ipc10/twisted-network-framework/index.html:    def connectionLost(self): self.peer.loseConnection();del self.peer
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:    def connectionLost(self):
./doc/historic/2004/ibm/talk.html:    def connectionLost(self):
./doc/historic/ipc10paper.html:    def connectionLost(self): self.peer.loseConnection();del self.peer
./doc/core/howto/tutorial/listings/finger/fingerproxy.py:    def connectionLost(self):
./doc/core/examples/chatserver.py:    def connectionLost(self):
```

These need to be audited and acknowledged as either being okay (for IMessage) or insane (and fixed).

It would be good to have unit tests for the examples too.

<details><summary>Searchable metadata</summary>

```
trac-id__1818 1818
type__defect defect
reporter__jerub jerub
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1150262718000000 1150262718000000
changetime__1150416741000000 1150416741000000
version__None None
owner__jerub jerub

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @jerub|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub commented|
|-|-|

These lines are in papers submitted to conferences, not sure we should 'fix' them:
```
./doc/historic/2002/ipc10/twisted-network-framework/index.html:    def connectionLost(self): self.peer.loseConnection();del self.peer
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self): self.factory.setUser(*self.lines)
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:      def connectionLost(self):
./doc/historic/2003/haifux/haifux.html:    def connectionLost(self):
./doc/historic/2004/ibm/talk.html:    def connectionLost(self):
./doc/historic/ipc10paper.html:    def connectionLost(self): self.peer.loseConnection();del self.peer
```

IMessage.connectionLost doesn't have a reason:
```
./twisted/mail/test/test_mail.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/alias.py:    def connectionLost(self):
./twisted/mail/mail.py:    def connectionLost(self):
./twisted/mail/smtp.py:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-5.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-6.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-7.tac:    def connectionLost(self):
./doc/mail/tutorial/smtpserver/smtpserver-8.tac:    def connectionLost(self):
./doc/mail/examples/emailserver.tac:    def connectionLost(self):
```

These two are now fixed (r17201):
```
./doc/core/howto/tutorial/listings/finger/fingerproxy.py:    def connectionLost(self):
./doc/core/examples/chatserver.py:    def connectionLost(self):
```

This code needs a unit test, and to be fixed, I haven't done it yet.
```
./twisted/words/im/proxyui.py:    def connectionLost(self):
```

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub commented|
|-|-|

branch is named: missing-reason-1818

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub commented|
|-|-|

This line of code is in a file with 26 lines and about 5 bugs and no unit tests.
```
./twisted/words/im/proxyui.py:    def connectionLost(self):
```
So i'm opening a new ticket for that.

I'm going to submit missing-reason-1818 for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Review tickets should be highest priority and unowned until someone takes responsibility for reviewing them.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @jerub|
|-|-|

Ok, next time don't take so long getting someone to review such a trivial change :P

**Comment by twisted-trac** [comment]
|<img alt="stephen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|stephen set status to `closed`|
|-|-|

(In [17230]) closes [#1818](https://github.com/twisted/twisted/issues/1818)
